### PR TITLE
Ensure Lightning is tested as RC/RC

### PR DESCRIPTION
**Pull request by mlxd**
Details:
Lightning is currently tested as RC PennyLane/Latest Lightning, which can diverge from the release-candidate for Lightning. This PR updates the RC table to have RC/RC and RC/latest categories.

### CODE REVIEW

**Comment by mlxd** [comment]
[sc-62303]

File: README.md [hunk 1]
```diff
@@ -132,16 +132,19 @@ The following table shows the current Catalyst compatibility with the Lightning
...
+|                                                                             |                                                                                                           Release candidate plugin / Release candidate PennyLane                                                                                                           |
+| :-------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Lightning](https://github.com/PennyLaneAI/pennylane-lightning)             | [![GitHub Workflow Status](https://img.shields.io/github/actions/workflow/status/PennyLaneAI/plugin-test-matrix/lightning-rc-rc.yml)](https://github.com/PennyLaneAI/plugin-test-matrix/actions/workflows/lightning-rc-rc.yml)           |
```
Comment (AmintorDusko) [inline README.md:137]: We can use the script we already have in the repository for that.
I am shamelessly stealing it from @vincentmr's [branch](https://github.com/PennyLaneAI/plugin-test-matrix/blob/0ade9615e6a46d2896c27de0192cc46b2dc0a451/README.md#feature-freeze-testing-matrix).
```suggestion
| [![GitHub Workflow Status](https://img.shields.io/github/actions/workflow/status/PennyLaneAI/pennylane-lightning/compat-check-release-release.yml?branch=master)](https://github.com/PennyLaneAI/pennylane-lightning/actions/workflows/compat-check-release-release.yml)          |
```

Comment (vincentmr) [reply README.md:137]: Thanks @AmintorDusko . Yes, we already have a kron job which also tests L-GPU (+MPI) and L-Kokkos (CUDA), so I think this is preferable.

Comment (vincentmr) [reply README.md:137]: If you agree @mlxd , I'll just open a PR trying to merge my branch.

Comment (mlxd) [reply README.md:137]: Whatever solution helps resolve this, I'm in favour of. But let's get it in before end of day if possible. If it'll take longer, let's go ahead with this one. We should also make sure to keep the split in the RC/RC from RC/latest this introduces.

**Pull request closed without merging**